shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.mm

```diff
@@ -4,6 +4,47 @@
 
 #import "flutter/shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.h"
 
+/// A proxy object that behaves like NSData represented in a Mapping.
+/// This isn't a subclass of NSData because NSData is in a class cluster.
+@interface FlutterMappingData : NSObject
+@end
+
+@implementation FlutterMappingData {
+  NSData* _data;
+  std::unique_ptr<fml::Mapping> _mapping;
+}
+
++ (NSData*)dataWithMapping:(std::unique_ptr<fml::Mapping>)mapping {
+  return (NSData*)[[[FlutterMappingData alloc] initWithMapping:std::move(mapping)] autorelease];
+}
+
+- (instancetype)initWithMapping:(std::unique_ptr<fml::Mapping>)mapping {
+  self = [super init];
+  if (self) {
+    const void* rawData = mapping->GetMapping();
+
+    // Const cast is required because the NSData API requires it despite
+    // guarentees that the buffer won't be deleted or modified.
+    _data = [[NSData alloc] initWithBytesNoCopy:const_cast<void*>(rawData)
```

Comment:
This is an extra allocation that wouldn't be happening before in lieu of a copy.